src/material/datepicker/calendar-body.ts

```diff
@@ -151,7 +152,11 @@ export class MatCalendarBody implements OnChanges, OnDestroy, AfterViewChecked {
   /** Width of an individual cell. */
   _cellWidth: string;
 
-  constructor(private _elementRef: ElementRef<HTMLElement>, private _ngZone: NgZone) {
+  constructor(
+    private _elementRef: ElementRef<HTMLElement>,
+    private _ngZone: NgZone,
+    private _intl: MatDatepickerIntl,
```

Comment:
Since the calendar body is exposed as a public API, this will have to be marked as optional initially. Here's an example of how it should be done: https://github.com/angular/components/blob/main/src/cdk/table/table.ts#L518

Reply:
Or just use the new `inject()` style? I think that's probably what we want to start doing. As a reference, see https://github.com/angular/components/pull/24941

Reply:
Okay, I made it optional. (This commit is on another PR, #24950).

Would switching all of Datepicker to the new `inject()` style be considered a breaking change?

Reply:
Not necessarily, but the shape of the constructor would have to stay the same. You can try changing it to something like:

```
constructor(...args: any[]) {}
```